script/tool/lib/src/publish_check_command.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:colorize/colorize.dart';
+import 'package:file/file.dart';
+import 'package:pubspec_parse/pubspec_parse.dart';
+
+import 'common.dart';
+
+class PublishCheckCommand extends PluginCommand {
+  PublishCheckCommand(
+    Directory packagesDir,
+    FileSystem fileSystem, {
+    ProcessRunner processRunner = const ProcessRunner(),
+  }) : super(packagesDir, fileSystem, processRunner: processRunner);
+
+  @override
+  final String name = 'publish-check';
+
+  @override
+  final String description =
+      'Checks to make sure that a plugin *could* be published.';
+
+  @override
+  Future<Null> run() async {
+    checkSharding();
+    final List<Directory> failedPackages = <Directory>[];
+
+    await for (Directory plugin in getPlugins()) {
+      if (!(await passesPublishCheck(plugin))) failedPackages.add(plugin);
+    }
+
+    if (failedPackages.isNotEmpty) {
+      final String error =
+          'FAIL: The following ${failedPackages.length} package(s) failed the '
+          'publishing check:';
+      final String joinedFailedPackages = failedPackages.join('\n');
+
+      final Colorize colorizedError = Colorize('$error\n$joinedFailedPackages')
+        ..red();
+      print(colorizedError);
+      throw ToolExit(64);
```

Comment:
Why 64 (vs just 1)?

Reply:
It's been awhile, but I believe it was out of habit from reading something like https://www.freebsd.org/cgi/man.cgi?query=sysexits. I changed it to 1.